pkg/flag/configs.go

```diff
@@ -0,0 +1,55 @@
+package flag
+
+import (
+	"context"
+	"encoding/json"
+	"log/slog"
+
+	"github.com/pkg/errors"
+	"github.com/torchiaf/kubectl-rancherx/pkg/log"
+	"k8s.io/helm/pkg/strvals"
+)
+
+func MergeValues(ctx context.Context, obj any, cfg *CommonConfig) error {
```

Comment:
Ok, now I can probably see why the original need for the `T` parameter. You can probably have it and make this typesafe.

To be more explicit I would change a bit the signature, and the behavior of this func. It does not need a `CommonConfig` and this can be simplified.

I was thinking something like:

```
MergeValues[T any](orig T, set []String) (T, error)
```

Also the fact that the logger needs the context sounds a bit strange to me.